### ISSUE TITLE
pylint json report doesn't have all the errors like in the console output

### ISSUE THREAD

**Issue by davidbluevine** [pylint-dev/pylint#8637]
### Bug description

Hi,
i am running pylint with pre-commit.
when the output is directed json report I see a subset of the errors that I see when the output is on the console output.
pylint version : 2.13.9

### Configuration

```ini
pre-commit files:
repos:
- repo: https://github.com/pycqa/pylint
  rev: v2.13.9
  hooks:
  - id: pylint
    name: pylint
    entry: pylint
    language: system
    types: [python]
    args:
      [
        "-rn", # Only display messages
        "--rcfile=pyproject.toml", "--output-format=json:my-report.json"
      ]
```

### Command used

```shell
pre-commit run --show-diff-on-failure --color=always --all-files
```

### Pylint output

```shell
.
```

### Expected behavior

the output on the console should have the same number of errors like in the json report.

### Pylint version

```shell
2.13.9
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. Could you provide a reproducer and maybe use the latest pylint (3.0.0a6) ? ``2.13.9`` is old at this point :)

**Comment by davidbluevine** [comment]
I can share the output if that you mean by reproducer.

**Comment by davidbluevine** [comment]
I see no one was assigned.

**Comment by Pierre-Sassoulas** [comment]
We're volunteer working on this on our free time (unless it's a security issue then it can go through Tidelift, or if you're sponsoring one of the maintainer https://github.com/sponsors/DanielNoord). So there's no automatic assignment in issues. If no one want to work on it, it's not going to be fixed. In fact the first step would be to create a reproducer, as this issue is very unlikely to be fixed without one. You're the more likely to be able to do that, do you want me to assign you to the issue ?

**Comment by jacobtylerwalls** [comment]
> I can't share the output if that you mean by reproducer. the issue is hat I get less errors in the json report compared to the console output. I changed to 3.0.0a6 and still the same issue.
cat report.json | grep .py | nl. the result -> 1459 lines
for the console output: pre-commit run --show-diff-on-failure --color=always --all-files | grep .py -> the result is 4212 lines

With a difference this large, you should be able to select only one out of these many differences, and rewrite it as a benign example having nothing to do with your closed-source codebase. Until then, will have to close this as not actionable.
